### PR TITLE
feat(std): support bench API's only flag

### PR DESCRIPTION
**Pull request by jopemachine**
## Related issue
#8041 

## Test
Manual Test

Here is the test code.
```ts
// benchTest.ts
import {
  bench,
  BenchmarkRunError,
  BenchmarkRunProgress,
  clearBenchmarks,
  ProgressState,
  runBenchmarks,
} from "./std/testing/bench.ts";

bench({
  name: "runs100ForIncrementX1e6",
  runs: 100,
  func(b: any): void {
    b.start();
    for (let i = 0; i < 1e6; i++);
    b.stop();
  },
});

bench({
  name: "runs100ForIncrementX1e6-2",
  only: true,
  runs: 100,
  func(b: any): void {
    b.start();
    for (let i = 0; i < 1e6; i++);
    b.stop();
  },
});

bench({
  name: "runs100ForIncrementX1e6-3",
  runs: 100,
  func(b: any): void {
    b.start();
    for (let i = 0; i < 1e6; i++);
    b.stop();
  },
});

await runBenchmarks({ silent: false });

```

Here is the result (of `deno run benchTest.ts`)

```
Check file:///Users/igyubong/Desktop/deno/benchTest.ts
running 1 benchmark ...
benchmark runs100ForIncrementX1e6-2 ... 
    100 runs avg: 0.3ms
benchmark result: DONE. 1 measured; 2 filtered
```

As expected, only items with the flag `only` are true were executed.

## Notes

1. I'm a newbie to `deno` community. I would appreciate it if you could let me know if there is anything out of the coding standard of `deno` or anything I missed :)

2. I haven't removed the existing code's `option` argument using Regexp. Please let me know if we need to remove this argument.

### CODE REVIEW

**Review by crypt096 — APPROVED** [review]

**Comment by littletof** [comment]
Thanks @jopemachine!
The basic feature looks good, I just have some nitpicks.
Other than those, and depending on the response on this https://github.com/denoland/deno/issues/8041#issuecomment-727209105, I think its good to go.

File: std/testing/bench.ts
Comment (littletof) [inline std/testing/bench.ts:26]: Since `only` is optional, it would be better to move it down below `runs?`.

File: std/testing/bench.ts
Comment (littletof) [inline std/testing/bench.ts:21]: Since `runBenchmarks` has an `only: Regexp` option too, I think it would be good to add JSDocs to these new `only: boolean` properties, so its clear what their function is.

File: std/testing/bench.ts
Comment (littletof) [inline std/testing/bench.ts:201]:
```suggestion
  // If there are benches where only is true, proceed to benchmark only those items
```

**Comment by littletof** [comment]
Also, could you please add tests that cover these different cases? (simple benches, without `skip` and `only` regexp, and an other, where `skip` and `only` regexp is also set in `runBenchmarks`)

Something like this: https://github.com/denoland/deno/blob/master/std/testing/bench_test.ts#L200-L216

**Comment by jopemachine** [comment]
Ok, I will. Thank you for your detailed review! :)

**Comment by jopemachine** [comment]
> Also, could you please add tests that cover these different cases? (simple benches, without `skip` and `only` regexp, and an other, where `skip` and `only` regexp is also set in `runBenchmarks`)
> 
> Something like this: https://github.com/denoland/deno/blob/master/std/testing/bench_test.ts#L200-L216

Ok, I will. Thank you for your detailed review! :)

**Comment by bartlomieju** [comment]
@littleof would you mind giving this PR another review? @jopemachine is it ready to land?

**Comment by jopemachine** [comment]
> @littleof would you mind giving this PR another review? @jopemachine is it ready to land?

@bartlomieju littleof left a suggestion on here (https://github.com/denoland/deno/issues/8041#issuecomment-727275760)

I think maybe it could be better to check the suggestion before landing.

**Comment by bartlomieju** [comment]
> > @littleof would you mind giving this PR another review? @jopemachine is it ready to land?
> 
> @bartlomieju littleof left a suggestion on here ([#8041 (comment)](https://github.com/denoland/deno/issues/8041#issuecomment-727275760))
> 
> I think maybe it could be better to check the suggestion before landing.

@jopemachine I think that suggestions makes a lot of sense, the more similar test and bench APIs are the better.

**Comment by jopemachine** [comment]
> > > @littleof would you mind giving this PR another review? @jopemachine is it ready to land?
> > 
> > 
> > @bartlomieju littleof left a suggestion on here ([#8041 (comment)](https://github.com/denoland/deno/issues/8041#issuecomment-727275760))
> > I think maybe it could be better to check the suggestion before landing.
> 
> @jopemachine I think that suggestions makes a lot of sense, the more similar test and bench APIs are the better.

@bartlomieju So, May I ask what should I do?
Is it alright to implement throwing an exception when the `filtered` result is not zero?
Or just indicating it in the output is better?

**Comment by bartlomieju** [comment]
@jopemachine I think throwing is preferable similar to how test runner exists with code 1 if there are tests with "only" option
https://github.com/denoland/deno/blob/27dd78601669a92d0200ad197adcf0919fdc9e46/cli/rt/40_testing.js#L338-L340

**Comment by jopemachine** [comment]
@bartlomieju,

https://github.com/denoland/deno/issues/8041#issuecomment-737297916

Could you review his comment about this issue?

If you decide after the review of the comment, I will start working on this issue again, thanks!

**Comment by bartlomieju** [comment]
@jopemachine I'm not really sure anymore what we should do... Let's wait a bit and discuss this issue during STD WG meeting

**Comment by bartlomieju** [comment]
@jopemachine we won't land this PR before standard library migration to a separate repository (https://github.com/denoland/deno/issues/9029#issuecomment-766934654). 

I'm still very much interested in discussing these changes on upcoming meetings and potentially landing the PR.

**Comment by bartlomieju** [comment]
Deno standard library was moved to [denoland/deno_std](https://github.com/denoland/deno_std), please reopen your PR there.

**Pull request closed without merging**
